File: step_definitions/Functionality Check on Outdated Browsers.java
package com.cucumber.steps;

import com.page_objects.OutdatedBrowserPage;
import io.cucumber.java.en.Given;
import io.cucumber.java.en.Then;
import io.cucumber.java.en.When;

public class OutdatedBrowserSteps extends OutdatedBrowserPage {

    @Given("the application is deployed on the stage environment")
    public void theApplicationIsDeployedOnTheStageEnvironment() {
        deployApplicationOnStage();
    }

    @Given("the application is opened in Internet Explorer 11")
    public void theApplicationIsOpenedInInternetExplorer11() {
        openApplicationInBrowser("Internet Explorer 11");
    }

    @Given("the application is opened in Firefox 45")
    public void theApplicationIsOpenedInFirefox45() {
        openApplicationInBrowser("Firefox 45");
    }

    @Given("the application is opened in Chrome 49")
    public void theApplicationIsOpenedInChrome49() {
        openApplicationInBrowser("Chrome 49");
    }

    @Given("the application is opened in an outdated browser")
    public void theApplicationIsOpenedInAnOutdatedBrowser() {
        openApplicationInOutdatedBrowser();
    }

    @When("the application is opened in Internet Explorer 11")
    public void whenApplicationIsOpenedInInternetExplorer11() {
        openApplicationInBrowser("Internet Explorer 11");
    }

    @When("navigating through the homepage")
    public void navigatingThroughTheHomepage() {
        navigateThroughHomepage();
    }

    @When("clicking on various links and buttons")
    public void clickingOnVariousLinksAndButtons() {
        clickLinksAndButtons();
    }

    @When("repeating steps 1-3")
    public void repeatingSteps1To3() {
        repeatSteps1To3();
    }

    @When("navigating through the application")
    public void navigatingThroughTheApplication() {
        navigateThroughApplication();
    }

    @When("verifying CSS styles")
    public void verifyingCSSStyles() {
        verifyCSSStyles();
    }

    @When("testing form submissions")
    public void testingFormSubmissions() {
        testFormSubmissions();
    }

    @When("checking for responsive design elements")
    public void checkingForResponsiveDesignElements() {
        checkResponsiveDesignElements();
    }

    @When("testing media content loading")
    public void testingMediaContentLoading() {
        testMediaContentLoading();
    }

    @When("verifying interactive elements")
    public void verifyingInteractiveElements() {
        verifyInteractiveElements();
    }

    @When("checking for deprecated HTML/CSS features")
    public void checkingForDeprecatedHTMLCSSFeatures() {
        checkDeprecatedHTMLCSSFeatures();
    }

    @When("testing accessibility features")
    public void testingAccessibilityFeatures() {
        testAccessibilityFeatures();
    }

    @When("verifying JavaScript functions")
    public void verifyingJavaScriptFunctions() {
        verifyJavaScriptFunctions();
    }

    @When("checking for browser-specific warnings or suggestions")
    public void checkingForBrowserSpecificWarningsOrSuggestions() {
        checkBrowserSpecificWarnings();
    }

    @Then("the application loads with functionality issues")
    public void theApplicationLoadsWithFunctionalityIssues() {
        verifyFunctionalityIssues();
    }

    @Then("homepage elements are not displayed correctly")
    public void homepageElementsAreNotDisplayedCorrectly() {
        verifyHomepageElementsDisplay();
    }

    @Then("some links and buttons do not function as expected")
    public void someLinksAndButtonsDoNotFunctionAsExpected() {
        verifyLinksAndButtonsFunctionality();
    }

    @Then("the application functions incorrectly with UI distortions")
    public void theApplicationFunctionsIncorrectlyWithUIDistortions() {
        verifyUIDistortions();
    }

    @Then("critical console errors are present")
    public void criticalConsoleErrorsArePresent() {
        verifyConsoleErrors();
    }

    @Then("CSS styles are inconsistent across outdated browsers")
    public void cssStylesAreInconsistentAcrossOutdatedBrowsers() {
        verifyCSSInconsistencies();
    }

    @Then("forms do not submit successfully")
    public void formsDoNotSubmitSuccessfully() {
        verifyFormSubmission();
    }

    @Then("responsive design elements do not adjust correctly")
    public void responsiveDesignElementsDoNotAdjustCorrectly() {
        verifyResponsiveDesignAdjustment();
    }

    @Then("media content does not load correctly")
    public void mediaContentDoesNotLoadCorrectly() {
        verifyMediaContentLoading();
    }

    @Then("interactive elements do not function correctly")
    public void interactiveElementsDoNotFunctionCorrectly() {
        verifyInteractiveElementsFunctionality();
    }

    @Then("the application relies on deprecated features")
    public void theApplicationReliesOnDeprecatedFeatures() {
        verifyDeprecatedFeatures();
    }

    @Then("accessibility features are not functional")
    public void accessibilityFeaturesAreNotFunctional() {
        verifyAccessibilityFeatures();
    }

    @Then("JavaScript functions do not execute correctly")
    public void javascriptFunctionsDoNotExecuteCorrectly() {
        verifyJavaScriptExecution();
    }

    @Then("browser-specific warnings are shown")
    public void browserSpecificWarningsAreShown() {
        verifyBrowserSpecificWarnings();
    }
}

File: step_definitions/Functionality Check on Outdated Browsers.js
import { Given, When, Then } from "cypress-cucumber-preprocessor/steps";
import OutdatedBrowserPage from "../pages/outdatedBrowserPage";

Given("the application is deployed on the stage environment", () => {
 OutdatedBrowserPage.visitStage();
});

Given("the application is opened in Internet Explorer 11", () => {
 OutdatedBrowserPage.openInIE11();
});

Given("the application is opened in Firefox 45", () => {
 OutdatedBrowserPage.openInFirefox45();
});

Given("the application is opened in Chrome 49", () => {
 OutdatedBrowserPage.openInChrome49();
});

Given("the application is opened in an outdated browser", () => {
 OutdatedBrowserPage.openInOutdatedBrowser();
});

When("the application is opened in Internet Explorer 11", () => {
 OutdatedBrowserPage.checkFunctionalityIssues();
});

When("navigating through the homepage", () => {
 OutdatedBrowserPage.navigateHomepage();
});

When("clicking on various links and buttons", () => {
 OutdatedBrowserPage.clickLinksAndButtons();
});

When("repeating steps 1-3", () => {
 OutdatedBrowserPage.repeatSteps();
});

When("navigating through the application", () => {
 OutdatedBrowserPage.navigateApplication();
});

When("verifying CSS styles", () => {
 OutdatedBrowserPage.verifyCSSStyles();
});

When("testing form submissions", () => {
 OutdatedBrowserPage.testFormSubmissions();
});

When("checking for responsive design elements", () => {
 OutdatedBrowserPage.checkResponsiveDesign();
});

When("testing media content loading", () => {
 OutdatedBrowserPage.testMediaContent();
});

When("verifying interactive elements", () => {
 OutdatedBrowserPage.verifyInteractiveElements();
});

When("checking for deprecated HTML/CSS features", () => {
 OutdatedBrowserPage.checkDeprecatedFeatures();
});

When("testing accessibility features", () => {
 OutdatedBrowserPage.testAccessibilityFeatures();
});

When("verifying JavaScript functions", () => {
 OutdatedBrowserPage.verifyJavaScriptFunctions();
});

When("checking for browser-specific warnings or suggestions", () => {
 OutdatedBrowserPage.checkBrowserWarnings();
});

Then("the application loads with functionality issues", () => {
 OutdatedBrowserPage.verifyFunctionalityIssues();
});

Then("homepage elements are not displayed correctly", () => {
 OutdatedBrowserPage.verifyHomepageElements();
});

Then("some links and buttons do not function as expected", () => {
 OutdatedBrowserPage.verifyLinksAndButtons();
});

Then("the application functions incorrectly with UI distortions", () => {
 OutdatedBrowserPage.verifyUIDistortions();
});

Then("critical console errors are present", () => {
 OutdatedBrowserPage.verifyConsoleErrors();
});

Then("CSS styles are inconsistent across outdated browsers", () => {
 OutdatedBrowserPage.verifyCSSInconsistencies();
});

Then("forms do not submit successfully", () => {
 OutdatedBrowserPage.verifyFormSubmissions();
});

Then("responsive design elements do not adjust correctly", () => {
 OutdatedBrowserPage.verifyResponsiveDesign();
});

Then("media content does not load correctly", () => {
 OutdatedBrowserPage.verifyMediaContentLoading();
});

Then("interactive elements do not function correctly", () => {
 OutdatedBrowserPage.verifyInteractiveElementsFunctionality();
});

Then("the application relies on deprecated features", () => {
 OutdatedBrowserPage.verifyDeprecatedFeatures();
});

Then("accessibility features are not functional", () => {
 OutdatedBrowserPage.verifyAccessibilityFeatures();
});

Then("JavaScript functions do not execute correctly", () => {
 OutdatedBrowserPage.verifyJavaScriptExecution();
});

Then("browser-specific warnings are shown", () => {
 OutdatedBrowserPage.verifyBrowserWarnings();
});